automata/fa/nfa.py

```diff
@@ -634,3 +634,56 @@ def transition(states_pair, symbol):
                     pair_stack.append((r_1, r_2))
 
         return True
+
+    @classmethod
+    def levenshtein(cls, input_symbols, query_string, D):
```

Comment:
@eliotwrobson It may be the conventional variable name for this algorithm, but I instead would prefer a snake_case variant like `distance`.

Reply:
@eliotwrobson Also, can we rename `query_string` to `input_str` if that semantically makes sense?

Finally, What do you think about keyword-only arguments for everything after (not including) `input_symbols`?

Reply:
I think that keyword-only arguments have only been used for input parameters that have default values set (and these don't have any sensible defaults). Though I think that the other renames are good.

Reply:
@caleb531 actually, I don't think that renaming `query_string` to `input_str` is a great idea, since the second name is used to refer to the input given to the NFA (whereas this is the string that the construction is based on).

Reply:
Perhaps `reference_string` could work.
Default value could be either `0` or `1` but I feel like it should always be specified so there should not be any default.
I think `max_distance` or `max_edit_distance` would be clearer than just `distance`.

Reply:
Made the variable name changes you suggested @Tagl . Also I agree, the user should really specify the distance.